components/eventing-controller/pkg/backend/nats/jetstream/jetstream.go

```diff
@@ -174,15 +175,25 @@ func (js *JetStream) SyncSubscription(subscription *eventingv1alpha1.Subscriptio
 		return err
 	}
 
-	// check whether NATS Subscription(s) were created for all the Kyma Subscription filters
+	if err := js.checkNATSSubscriptionsCount(subscription); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// checkNATSSubscriptionsCount checks whether NATS Subscription(s) were created for all the Kyma Subscription filters
+func (js *JetStream) checkNATSSubscriptionsCount(subscription *eventingv1alpha1.Subscription) error {
 	for _, subject := range subscription.Status.CleanEventTypes {
 		jsSubject := js.GetJetStreamSubject(subject)
 		jsSubKey := NewSubscriptionSubjectIdentifier(subscription, jsSubject)
 		if _, ok := js.subscriptions[jsSubKey]; !ok {
-			return errors.New(NoNatsSubscriptionErr + jsSubKey.NamespacedName())
+			return errors.Errorf(NoNatsSubscriptionErrMask, subject)
 		}
 	}
-
+	if len(subscription.Status.CleanEventTypes) > 0 && len(js.subscriptions) == 0 {
```

Comment:
what is there is mismatch? 
eg. len(subscription.Status.CleanEventTypes) = 5 and len(js.subscriptions) = 3

Reply:
Improved the condition